src/libsync/propagateupload.cpp

```diff
@@ -542,6 +542,15 @@ QMap<QByteArray, QByteArray> PropagateUploadFileCommon::headers()
 
 void PropagateUploadFileCommon::finalize()
 {
+    // Update the quota, if known
+    if (!_quotaUpdated) {
+        auto quotaIt = propagator()->_folderQuota.find(QFileInfo(_item->_file).path());
+        if (quotaIt != propagator()->_folderQuota.end()) {
+            quotaIt.value() -= _item->_size;
+        }
+        _quotaUpdated = true;
+    }
```

Comment:
What is this change? What's the lifetime of PropagateUploadFileCommon objects? Could this cause issues because of missing cache invalidation?

Reply:
See the commit message.
See the code in the next line, we might call the same function again if we where lacking permissions, which basically are always with an oc10 server.